pandas/tseries/tests/test_timeseries.py

```diff
@@ -772,6 +772,48 @@ def test_to_datetime_unit(self):
         result = to_datetime([1, 2, 111111111], unit='D', errors='coerce')
         tm.assert_index_equal(result, expected)
 
+    def test_to_datetime_origin(self):
+        units = ['D', 's', 'ms', 'us', 'ns']
+        # Addresses Issue Number 11276, 11745
+        # for origin as julian
+        julian_dates = pd.date_range(
+            '2014-1-1', periods=10).to_julian_date().values
+        result = Series(pd.to_datetime(
+            julian_dates, unit='D', origin='julian'))
+        expected = Series(pd.to_datetime(
+            julian_dates - pd.Timestamp(0).to_julian_date(), unit='D'))
+        assert_series_equal(result, expected)
```

Comment:
add tests for invalid when ``origin='julian'``. IOW non-integer, out-of-range julian dates.